src/replication.c

```diff
@@ -2252,7 +2260,14 @@ void readSyncBulkPayload(connection *conn) {
 
         serverLog(LL_NOTICE, "PRIMARY <-> REPLICA sync: Loading DB in memory");
         startLoading(server.repl_transfer_size, RDBFLAGS_REPLICATION, asyncLoading);
-
+        if (replicationSupportBypassCRC(conn, use_diskless_load, usemark)) {
+            /* We can bypass CRC checks when data is transmitted through a verified stream.
+             * The usemark flag indicates that the primary is streaming the data directly without
+             * writing it to storage.
+             * Similarly, the use_diskless_load flag indicates that the
+             * replica will load the payload directly into memory without first writing it to disk. */
```

Comment:
This comment is documenting the internal behavior of the function, which means if the function changes this comment is unlikely to get updated. I would move the comment to `replicationSupportBypassCRC`.